R/FilterState.R

```diff
@@ -582,32 +597,85 @@ FilterState <- R6::R6Class( # nolint
       values
     },
 
-    # Disables `FilterState`
-    # `state` is moved to cache and set to `NULL`
+    # Disables this `FilterState`.
+    #
+    # Filter state properties are packed into a `teal_slice` and saved in cache (private field).
+    # State properties that refer to selection are set to NULL.
+    #
     # @return `NULL` invisibly
     disable = function() {
-      private$cache <- self$get_state()
+      logger::log_trace("{ class(self)[1] }$set_state disabling fiter state of variable: { private$varname }")
+
+      private$cache_state()
```

Comment:
It seems like caching and restoring doesn't really make sense. Instead of all these methods we can just add condition in `get_call`. So:

- when disabled returned call is NULL so the data won't be filtered
- when disabled returned state will be returned with all fields but with flag `disabled = TRUE`

Reply:
I will overhaul disabling/enabling in #206, this is getting out of scope.

Reply:
#206 felt like very UI driven to me which would still be out of scope since the changes is more around back-end functionality. I suggest we create a new issue instead if the suggested update is outside of the scope.

Reply:
Oh, #206 is most certainly back-end driven. In the current implementation `filter_state_disable` caches and deletes all filter states, and `filter_state_enable` re-instantiates filter states based on the cached properties.